imgui-examples/examples/id_wrangling.rs

```diff
@@ -0,0 +1,45 @@
+mod support;
+
+fn main() {
+    let system = support::init(file!());
+    system.main_loop(move |_, ui| {
+        let items = vec!["a", "b", "c", "d"];
+
+        ui.window("Broken Example")
+            .position([0.0, 0.0], imgui::Condition::FirstUseEver)
+            .size([390.0, 200.0], imgui::Condition::FirstUseEver)
+            .build(|| {
+                ui.text("Broken! Only first button responds to clicks");
+
+                // Because all our buttons have the same label (and thus ID),
+                // only the first button responds to clicks!
+                for it in &items {
+                    ui.text(it);
+                    for num in 0..5 {
+                        ui.same_line();
+                        if ui.button("Example") {
+                            println!("{}: {}", it, num);
+                        }
+                    }
+                }
+            });
+
+        ui.window("Good Example")
+            .position([400.0, 0.0], imgui::Condition::FirstUseEver)
+            .size([390.0, 200.0], imgui::Condition::FirstUseEver)
+            .build(|| {
+                ui.text("Works!");
+                for it in &items {
+                    let _label_id = ui.push_id(it);
```

Comment:
Hm, this usage suggests to me we should consider making the result of push_id deref into `ui`, so that you can do `let ui = ui.push_id(...);` (which avoids the footguns of `let _ = ui.push_id(...)` or forcing you to come up with a dummy name).

Reply:
Not only do I agree with that, I have been working on EXACTLY this change to the entire library.

this is the sort of thing that would be required to have proper mutability support for `Ui` (ie, some methods will take `&mut self` and others will take `&self`). For example, it occured to me that if the drop token for the table's api was also where you access all of the table's methods, then we solve https://github.com/imgui-rs/imgui-rs/issues/524 at the type level. 
There are issues and i'm not sure we'll get that complex to start with, but I'm in total agreement here on that idea